### ISSUE TITLE
`.rolling_exp` arguments could be clearer

### ISSUE THREAD

**Issue by max-sixty** [pydata/xarray#8123]
### Is your feature request related to a problem?

Currently we call `.rolling_exp` like:

```
da.rolling_exp(date=20).mean()
```

`20` refers to a "standard" window type — broadly "the same average distance as a simple rolling window. That works well, and matches the `.rolling(date=20).mean()` format. 

But we also have different window types, and this makes it a bit incongruent:

```
da.rolling_exp(date=0.5, window_type="alpha").mean()
```

...since the `window_type` is completely changing the meaning of the value we pass to the dimension argument. A bit like someone asking "how many apples would you like to buy", and replying "5", and then separately saying "when I said 5, I meant 5 _tonnes_".

### Describe the solution you'd like

One option would be:

```
.rolling_exp(dptr={"alpha": 0.5})
```

We pass a dict if we want a non-standard window type — so the value is attached to its type.

We could still have the original form for `da.rolling_exp(date=20).mean()`.

### Additional context

(I realize I wrote this originally, all criticism directed at me! This is based on feedback from a colleague, which on reflection I agree with.)

Unless anyone disagrees, I'll try and do this soon-ish™

**Comment by benbovy** [comment]
Although I'm not familiar with this, I have a couple of comments / questions:

- A tuple might be better than a dict to group a single window value with its type?
- How would it work for multi-dimensional windows with non-default window type? Would it make sense to provide different types for different dimensions? If not, would we still have to provide the value and the (redundant) type for each dimension?

**Comment by max-sixty** [comment]
> A tuple might be better than a dict to group a single window value with its type?

Yes, either seem reasonable to me!

> How would it work for multi-dimensional windows with non-default window type?

For rolling, we don't have multiple dimensions. (I guess you could imagine something which took "everything with a value of x less than 5 and y less than 10" at (5, 10), but I've never seen anything like this in the wild...)

**Comment by benbovy** [comment]
> For rolling, we don't have multiple dimensions

Ah yes I see. Yeah it is hard to imagine how multiple dimensions would work :). I was confused by the API, i.e., the `window` mapping and `**window_kwargs` (I'm not using rolling often). I understand that it is a common pattern in Xarray, but here it feels a bit weird to me using this pattern for a unique dimension / value.

**Comment by max-sixty** [comment]
Yes I very much agree in isolation. But then it means that it's inconsistent with `.shift(date=2)` (or `.diff` / `.reindex`, lots of methods...).

How do you think we balance those?

**Comment by benbovy** [comment]
Yes I also very much agree xarray wise, having a common signature pattern for all those methods certainly reduces cognitive load. That said, it is precisely because I use more often `.sel`, `.reindex`, `.diff`... than `.rolling` that I have just been fooled by the signature of the latter.

Ideally we should have two clear and distinct patterns for the two cases, i.e., for the methods accepting 0+ ('key', 'value') pairs vs. for the methods accepting only one ('key', 'value') pair, where 'key' is a dimension or a variable name. But I think that we are all used to the current API so we should probably keep the current way.

We shouldn't make it more complicated, though. Taking the examples,

```python
.rolling_exp(date=0.5)
.rolling_exp(date={"alpha": 0.5})
```

an annotated signature that would be fully consistent with the `either_dict_or_kwargs` pattern would look like,

```python
WindowType = Literal["span", "com", "halflife", "alpha"]

def rolling_exp(
    window: Mapping[Hashable, int | Mapping[WindowType, int | float]] | None = None,
    **window_kwargs
) -> RollingExp:
    ...
```

whereas we could just write it like

```python
def rolling_exp(
    dim: Hashable,
    window_value: int | float,
    window_type: WindowType = "span"
) -> RollingExp:
    ...

# or

def rolling_exp(
    dim: Hashable,
    window: int | tuple[int | float, WindowType],
) -> RollingExp:
    ...
```

**Comment by max-sixty** [comment]
Yes, I think that's very fair, well explained. I'm mostly persuaded!

(FYI `int | float` is the same as `float` IIRC!)
